crates/sui-types/src/execution.rs

```diff
@@ -167,6 +167,25 @@ impl ExecutionResultsV2 {
                 }
             }
 
+            // Record start version for ConsensusV2 objects.
+            if let Owner::ConsensusV2 { start_version, .. } = &mut obj.owner {
+                debug_assert!(!self.deleted_object_ids.contains(id));
+
+                if let Some(Owner::ConsensusV2 {
+                    start_version: previous_start_version,
+                    ..
+                }) = input_objects.get(id).map(|obj| &obj.owner)
+                {
+                    // Assign existing start_version in case a ConsensusV2 object was
+                    // re-transferred to the same Owner type.
```

Comment:
As you mentioned since you could be transferring it to another address but still in the form of ConsensusV2 type, "re-transferred" seems inaccurate here. (which applies better to the old shared object case)

Reply:
can you say more about the inaccuracy? in case you transfer to another SingleOwner address with ConsensusV2 type, we still want this condition to trigger and preserve the original `start_version`, so as far as I can tell this is accurate as written?

Reply:
would it make more sense if I just wrote "transferred" instead of "re-transferred"?

Reply:
changing to "transferred" as agreed on chat